WordPress/gutenberg
# Block Editor: Refactor `BlockControls` tests to `@testing-library/react`

## What?
We've recently started refactoring `enzyme` tests to `@testing-library/react`.

This PR refactors the `BlockControls` tests from `enzyme` to `@testing-library/react`.

## Why?
`@testing-library/react` provides a better way to write tests for accessible components that is closer to the way the user experiences them.

## How?
We're straightforwardly replacing `enzyme` tests with `@testing-library/react` ones, using `jest-dom` matchers and mocks to avoid testing unrelated implementation details. We're improving the test granularity by removing the snapshot and adding concrete assertions.

## Testing Instructions
Verify tests pass: `npm run test:unit packages/block-editor/src/components/block-controls/test/index.js`
